nixos/modules/services/misc/nix-daemon.nix

```
@@ -670,7 +680,7 @@ in
         concatMapStrings
           (machine:
             (concatStringsSep " " ([
-              "${optionalString (machine.sshUser != null) "${machine.sshUser}@"}${machine.hostName}"
+              "${machine.protocol}://${optionalString (machine.sshUser != null) "${machine.sshUser}@"}${machine.hostName}"
```

This kinda breaks my workflow. Could we do `
"${optionalString (machine.protocol != "ssh") "${machine.protocol}://"}${optionalString (machine.sshUser != null) "${machine.sshUser}@"}${machine.hostName}"` instead?

---

Prepending ``ssh://`` should be a no opt. Whats the problem you are encountering? Did you before prepend the protocol to the user? If so that is no longer supported and was a hack in the first place.

```suggestion
              "${optionalString (machine.protocol != "ssh") "${machine.protocol}://"}${optionalString (machine.sshUser != null) "${machine.sshUser}@"}${machine.hostName}"
```

---

I'm using a single aarch64 machine for hydra to build armv7l-linux binaries and it only works when I explicitly add localhost to `nix.buildMachines` with `armv7l-linux` in `systems`:
```
nix.buildMachines = [
  { hostName = "localhost";
    systems = [ "aarch64-linux" "armv7l-linux" ] ;
    supportedFeatures = [ "nixos-test" "big-parallel" "benchmark" ];
    maxJobs = 4;
  }
];
```
With this change the `/etc/nix/machines` now contains `ssh://localhost aarch64-linux,armv7l-linux - 4 1 nixos-test,big-parallel,benchmark  -` which doesn't work because ssh keys and stuff is not configured and I have to hack it like this to make hydra happy:
```
environment.etc."nix/machines".text = lib.mkForce "localhost aarch64-linux,armv7l-linux - 4 1 nixos-test,big-parallel,benchmark  -";
nix.buildMachines = [
  { hostName = "localhost";
    systems = [ "aarch64-linux" "armv7l-linux" ] ;
    supportedFeatures = [ "nixos-test" "big-parallel" "benchmark" ];
    maxJobs = 4;
  }
];
```

---

If you set this via binfmt and nix.settings this shouldn't be required to configure extra.

---

I set [extra-platforms](https://gitlab.com/misuzu/nixos-armv7l/-/blob/da05c11e50fde41c533c9f1c31927bb1e97e56e6/hosts/popuko/armv7l.nix#L16), but that's not enough for hydra. According to the [wiki](https://nixos.wiki/wiki/Hydra), supportedFeatures aren't picked up either.

---

hmmm, I workaround this in a hacky way I must admit. We should probably add null/empty string to the enum

---

#193009